dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/All.java

```diff
@@ -49,37 +50,61 @@ public void execute(final Namespace ns, final DockerfileGitHubUtil dockerfileGit
         }
         List<ProcessingErrors> imagesThatCouldNotBeProcessed = new LinkedList<>();
         AtomicInteger numberOfImagesToProcess = new AtomicInteger();
-        for (Map.Entry<String, JsonElement> imageToTag : imageToTagStore) {
-            numberOfImagesToProcess.getAndIncrement();
-            String image = imageToTag.getKey();
-            String tag = imageToTag.getValue().getAsString();
-            try {
-                PullRequests pullRequests = getPullRequests();
-                GitHubPullRequestSender pullRequestSender = getPullRequestSender(dockerfileGitHubUtil, ns);
-                GitForkBranch gitForkBranch = getGitForkBranch(image, tag, ns);
-
-                log.info("Finding Dockerfiles with the image name {}...", image);
+        Optional<Exception> failureMessage;
+        String store = ns.get(Constants.STORE);
+        URI storeUri = new URI(store);
+        ImageTagStore imageTagStore;
+        switch (storeUri.getScheme()) {
+            case (S3Store.s3Prefix):
+                log.info("The underlying data store is S3.");
+                AmazonS3 s3 = AmazonS3ClientBuilder.standard().withRegion(Regions.DEFAULT_REGION).build();
+                imageTagStore = new S3Store(s3, store);
+                break;
+            default:
+                log.info("The underlying data store is a Git repository.");
+                imageTagStore = this.dockerfileGitHubUtil.getGitHubJsonStore(store);
```

Comment:
This is a good use case for an Enum, create one value for S3 and value for Git.  Create a static method called `fromStoreUri` that produces an enum based upon the uri parameter.  Each enum value has a `getStore` method that returns the correct tag store implementation.